src/main/java/io/cdap/plugin/gcp/bigquery/sink/AbstractBigQuerySink.java

```diff
@@ -321,7 +319,28 @@ private Schema getTableSchema(
       LOG.info("Table [%s] doesn't have a schema. Using input schema for writing records.", tableName);
       return null;
     }
-    return BigQueryUtil.getTableSchema(bqSchema, collector);
+    return getOrderedSchema(bqSchema, configuredSchema);
```

Comment:
I think it's better for this call to happen outside of this method, in the overrideOutputSchemaWithTableSchemaIfNeeded method, and keep getTableSchema() the same so that it only fetches the table schema. Otherwise the contract is a bit confusing.